[bitnami-mysql] replication not working, authentication failed

### Name and Version

bitnami/mysql -- version: 9.0.4

### What steps will reproduce the bug?

Run helm chart with 

architecture: replication
rootPassword: "password"
database: "my_commondb"
username: "keysight"
password: "keysight"
replicationUser: replicator
replicationPassword: "keysight"
secondary-  replicaCount: 2


### Are you using any custom parameters or values?

_No response_

### What is the expected behavior?

_No response_

### What do you see instead?

k logs common-mysql-secondary-0
mysql 14:03:53.13 
mysql 14:03:53.13 Welcome to the Bitnami mysql container
mysql 14:03:53.13 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
mysql 14:03:53.13 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
mysql 14:03:53.14 
mysql 14:03:53.14 INFO  ==> ** Starting MySQL setup **
mysql 14:03:53.16 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mysql 14:03:53.16 INFO  ==> Initializing mysql database
mysql 14:03:53.17 WARN  ==> The mysql configuration file '/opt/bitnami/mysql/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mysql 14:03:53.18 INFO  ==> Using persisted data
mysql 14:03:53.20 INFO  ==> Running mysql_upgrade
mysql 14:03:53.20 INFO  ==> Starting mysql in background
find: '/docker-entrypoint-startdb.d/': No such file or directory
mysql 14:04:05.25 INFO  ==> Stopping mysql
mysql 14:04:09.28 INFO  ==> ** MySQL setup finished! **

mysql 14:04:09.31 INFO  ==> ** Starting MySQL **
2022-05-23T14:04:09.598975Z 0 [Warning] [MY-011068] [Server] The syntax 'log_slave_updates' is deprecated and will be removed in a future release. Please use log_replica_updates instead.
2022-05-23T14:04:09.598997Z 0 [Warning] [MY-011069] [Server] The syntax '--master-info-repository' is deprecated and will be removed in a future release.
2022-05-23T14:04:09.599010Z 0 [Warning] [MY-011069] [Server] The syntax '--relay-log-info-repository' is deprecated and will be removed in a future release.
2022-05-23T14:04:09.599128Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
2022-05-23T14:04:09.599154Z 0 [System] [MY-010116] [Server] /opt/bitnami/mysql/bin/mysqld (mysqld 8.0.29) starting as process 1
2022-05-23T14:04:09.601083Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
2022-05-23T14:04:09.601095Z 0 [Warning] [MY-013244] [Server] --collation-server: 'utf8_general_ci' is a collation of the deprecated character set UTF8MB3. Please consider using UTF8MB4 with an appropriate collation instead.
2022-05-23T14:04:09.607156Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2022-05-23T14:04:09.917433Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2022-05-23T14:04:10.131672Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2022-05-23T14:04:10.131727Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2022-05-23T14:04:10.164607Z 5 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
2022-05-23T14:04:10.168741Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 1 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:04:10.169387Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060, socket: /tmp/mysqlx.sock
2022-05-23T14:04:10.169475Z 0 [System] [MY-010931] [Server] /opt/bitnami/mysql/bin/mysqld: ready for connections. Version: '8.0.29'  socket: '/opt/bitnami/mysql/tmp/mysql.sock'  port: 3306  Source distribution.
2022-05-23T14:04:12.449851Z 14 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:04:20.171346Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 2 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:04:22.462482Z 15 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:04:30.173669Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 3 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:04:32.459500Z 16 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:04:40.176428Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 4 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:04:42.460751Z 17 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:04:50.178939Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 5 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:04:52.461829Z 18 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:05:00.181286Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 6 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:05:02.450396Z 19 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:05:10.183963Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 7 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:05:12.458195Z 20 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:05:20.186235Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 8 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:05:22.459347Z 21 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:05:30.188958Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 9 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
2022-05-23T14:05:32.466140Z 22 [Warning] [MY-013360] [Server] Plugin sha256_password reported: ''sha256_password' is deprecated and will be removed in a future release. Please use caching_sha2_password instead'
2022-05-23T14:05:40.191983Z 5 [ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to master 'replicator@common-mysql-primary:3306' - retry-time: 10 retries: 10 message: Authentication plugin 'caching_sha2_password' reported error: Authentication requires secure connection. Error_code: MY-002061


### Additional information

_No response_

--- a/bitnami/mysql/Chart.yaml
+++ b/bitnami/mysql/Chart.yaml
@@ -25,4 +25,4 @@ name: mysql
 sources:
   - https://github.com/bitnami/bitnami-docker-mysql
   - https://mysql.com
-version: 9.0.4
+version: 9.0.5

--- a/bitnami/mysql/templates/secrets.yaml
+++ b/bitnami/mysql/templates/secrets.yaml
@@ -16,6 +16,6 @@ data:
   mysql-root-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mysql-root-password" "length" 10 "providedValues" (list "auth.rootPassword") "context" $) }}
   mysql-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mysql-password" "length" 10 "providedValues" (list "auth.password") "context" $) }}
   {{- if eq .Values.architecture "replication" }}
-  mysql-replication-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mysql-replication-password" "length" 10 "providedValues" (list "auth.replicationPassword") "context" $) | b64enc | quote }}
+  mysql-replication-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mysql-replication-password" "length" 10 "providedValues" (list "auth.replicationPassword") "context" $) }}
   {{- end }}
 {{- end }}

